.eslintrc.json

```diff
@@ -36,7 +36,8 @@
       "files": ["*.stories.tsx", "packages/*/demo/**/*"],
       "rules": {
         "react/button-has-type": "off",
-        "react/no-array-index-key": "off"
+        "react/no-array-index-key": "off",
+        "func-name-matching": "off"
```

Comment:
From that link...

> To avoid linting issues, it is recommended to use a function with a capitalized name. If you are not concerned with linting, you may use an arrow function.

Are you sure they aren't referring to their own Storybook linter (which we don't use)? If so, then Garden would definitely prefer arrow functions.

Reply:
That applies to our eslint config as well. To be clear, we can still use arrow functions and should do so when not using hooks.

`func-name-matching` expects `{ render: function render {...} }`, however [react-hooks/rules-of-hooks](https://legacy.reactjs.org/docs/hooks-rules.html) requires that a function be `PascalCased` when using hooks. Disabling `react-hooks/rules-of-hooks` can lead to bugs but turning off `func-name-matching` won't. That's why I think it's a better choice. 

The alternative is to move that function elsewhere.

```
const Controlled: Story = {
  render: function Render (args) {...}
  ...
}
```
Becomes 

```
export const Controlled: Story = {
  render: Render,
  ...
}

function Render (args) {...}
```

But I think it's more cumbersome and would reduce readability.

Reply:
ic – yeah, then what you have is the right trade-off. Thanks for the explanation.